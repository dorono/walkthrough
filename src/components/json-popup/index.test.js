import React from 'react';
import {shallow} from 'enzyme';
import JsonPopup from './index';

const defaultProps = {
    onClose: () => null,
    show: false,
    data: JSON.stringify({test: ''}),
};

describe('JsonPopup', () => {
    it('should render without errors', () => {
        shallow(<JsonPopup {...defaultProps} />);
    });

    it('should match snapshot', () => {
        expect(shallow(<JsonPopup {...defaultProps} />)).toMatchSnapshot();
    });
});
