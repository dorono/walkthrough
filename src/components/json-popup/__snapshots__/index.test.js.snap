// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonPopup should match snapshot 1`] = `
<Modal
  className="modalRoot defaultPosition"
  setModalSize={false}
  show={false}
>
  <ModalHeader
    className="modalHeader"
  >
    ANCHOR - BITCOIN
  </ModalHeader>
  <ModalBody
    className="modalBody"
  >
    <div
      className="actionContainer"
    >
      <span>
        JSON
      </span>
      <CopyToClipboard
        text="{\\"test\\":\\"\\"}"
      >
        <span
          className="button buttonCopy"
          onClick={[Function]}
        >
          <span
            className="toast"
          >
            Copied
          </span>
          <img
            alt="Copy Content"
            src="test-file-stub"
          />
          Copy JSON
        </span>
      </CopyToClipboard>
    </div>
    <div
      className="jsonContainer"
    >
      <JsonViewer
        className="jsonRoot"
        data="{\\"data\\":\\"{\\\\\\"test\\\\\\":\\\\\\"\\\\\\"}\\"}"
      />
    </div>
  </ModalBody>
  <ModalFooter
    className="modalFooter"
  >
    <Button
      className="button cancel"
      id="cancel-settings-popup"
      key="cancel"
      onClick={[Function]}
      title="CLOSE"
    />
  </ModalFooter>
</Modal>
`;
